test/prototype_transforms_dispatcher_infos.py

```diff
@@ -15,11 +17,24 @@
 class DispatcherInfo:
     dispatcher: Callable
     kernels: Dict[Type, Callable]
-    skips: Sequence[Skip] = dataclasses.field(default_factory=list)
-    _skips_map: Dict[str, Skip] = dataclasses.field(default=None, init=False)
+    skips: Sequence = dataclasses.field(default_factory=list)
+    _skips_map: Dict = dataclasses.field(default=None, init=False)
+
+    test_marks: Sequence[TestMark] = dataclasses.field(default_factory=list)
+    _test_marks_map: Dict[str, List[TestMark]] = dataclasses.field(default=None, init=False)
 
     def __post_init__(self):
-        self._skips_map = {skip.test_name: skip for skip in self.skips}
+        test_marks_map = defaultdict(list)
+        for test_mark in self.test_marks:
+            test_marks_map[test_mark.test_id].append(test_mark)
+        self._test_marks_map = dict(test_marks_map)
```

Comment:
This is the same on `KernelInfo` and here on the `DispatcherInfo`. I'm going to factor this out into a common base class in a follow-up PR since this PR is already quite large.